controller/node_controller.go

```diff
@@ -364,6 +364,13 @@ func (nc *NodeController) syncNode(key string) (err error) {
 	log := getLoggerForNode(nc.logger, node)
 
 	if node.DeletionTimestamp != nil {
+		// If V2 engine is enabled, we must clean up SPDK-related drivers
+		// before deleting the node. This ensures that SPDK devices can be reused on future node joins.
+		if err := nc.cleanupDisksBeforeNodeDeletion(node); err != nil {
+			nc.logger.WithError(err).Errorf("Cleanup spdk disk driver")
+			return err
+		}
```

Comment:
Best-effort cleanup and don't hinder the node deletion. WDYT?

```suggestion
		if errCleanup := nc.cleanupDisksBeforeNodeDeletion(node); errCleanup != nil {
			nc.logger.WithError(errCleanup).Errorf("Cleanup spdk disk driver")
		}
```

Reply:
Sure, even though it fail, we have work around to handle it